js/script.js

```diff
@@ -1,290 +1,127 @@
-        if (cmte.members.length) {
-          sortedUpperOutput += '<li><a href="#"  data-cmte-id="' + cmte.id + '">' + cmte.committee + '</a></li>';
-        }
-      });
-      var sortedLowerOutput = "";
-      $.each(sortedLowerCommittees, function(key, cmte) {
-        if (cmte.members.length) {
-        sortedLowerOutput += '<li><a href="#"  data-cmte-id="' + cmte.id + '">' + cmte.committee + '</a></li>';
-        }
-      });
-      $('ul#tinyDropUpper')
-        .prepend(sortedUpperOutput);
-      $('ul#tinyDropLower')
-        .prepend(sortedLowerOutput);
-
-      var committee1 = getQueryVariable("committee");
-      if (committee1) {
-        $('[data-cmte-id="'+committee1+'"]').click();
+  openstates.getState(state).done(function(committeeList) {
+    //TODO: clean up some of these loops
+    var upper = [];
+    var lower = [];
+    committeeList.forEach(function(comm) {
+      if (comm.subcommittee) return; //reject subs and empties
+      if (comm.chamber == "upper") {
+        upper.push(comm);
       }
-    })
+      if (comm.chamber == "lower") {
+        lower.push(comm);
+      }
+    });
+    upper = _.sortBy(upper, "committee");
+    lower = _.sortBy(lower, "committee");
+    var upperHTML = templates.list({ committees: upper });
+    $("#tinyDropUpper").html(upperHTML);
+    var lowerHTML = templates.list({committees: lower});
+    $("#tinyDropLower").html(lowerHTML);
+    
+    var committee = getQueryVar("committee");
+    if (committee) {
+      $('[data-cmte-id="'+committee+'"]').click();
+    }
+  });
 };
+
 // Get committee detail
-var getCommitteeDetail = function(committee_id) {
-  var committeeRequest = $.ajax({
-    dataType: "json",
-    url: "http://openstates.org/api/v1/committees/" + committee_id + "/",
-    data: {
-      apikey: "9e3e71730ae34e1ebbf4dd0e1c346c07"
+var getCommitteeDetail = function(id) {
+  openstates.getCommittee(id).done(function(committee) {
+    if (committee.members.length > 0) {
+      addLegislators(committee);
+    } else {
+      $(".panel").html("<div id='no-reported'>This committee <br/>(" + committee.committee + ")<br/>has no reported members.</div>");
+      $(".entry").html("");
     }
-  })
-    .done(function(committee, textStatus, jqXHR) {
-//      var counter = 0;
-//      leg_id_array = _.pluck(committee.members, 'leg_id');
-//      console.log("memberIDs", leg_id_array);
-      if (committee.members.length > 0) {
-        console.log("getCommitteeDetail",committee);
-        addLegislators(committee);
-      } else {
-        $(".panel")
-          .html("<div id='no-reported'>This committee <br/>(" + committee.committee + ")<br/>has no reported members.</div>");
-        $(".entry")
-          .html("");
-      }
-    });
+  });
 };
+
 // append legislator detail to committee
 var addLegislators = function(committee) {
-  //  $('#update1-left pre').html("<h2>"+committee.committee+"</h2>");
+  //render all at once, not as it goes
```

Comment:
To clarify--I think you re-render in the code below as each API call for member data resolves. That's why you see them pop in out of order. If you wanted to wait for all of them, you could use `$.when` to run your code once all the API promises have been resolved:

```
var promises = committee.members.map(function(m) { return openstates.getMember(m.leg_id) };
var allDone = $.when.apply(null, promises); //collect responses
allDone.then(function() {
  var responses = arguments;
  responses.forEach(function(detail, i) {
    var member = committee.members[i]; //get the matching member data
    //process role, party
    member.detail = detail;
  });
  $("#committee-list").html( templates.member({ cmte: committee });
  currentCommittee = committee;
});